File: 2232-adding-spaces-to-a-string/2232-adding-spaces-to-a-string.java
class Solution {
    public String addSpaces(String s, int[] spaces) {
        int i = 0, k = 0;
        StringBuilder sb = new StringBuilder("");
        while (i < s.length() && k < spaces.length) {
            if (i == spaces[k]) {
                sb.append(" ");
                k++;
            }
            sb.append(s.charAt(i));
            i++;
        }
        
        while (i < s.length())
            sb.append(s.charAt(i++));
        return sb.toString();
    }
}

File: 2232-adding-spaces-to-a-string/2232-adding-spaces-to-a-string.js
/**
 * @param {string} s
 * @param {number[]} spaces
 * @return {string}
 */
var addSpaces = function(s, spaces) {
    let words = [], endIdx = spaces[0], startIdx = 0;
    
    for (let i = 0; i <= spaces.length; i++) {
        let word = s.slice(startIdx, endIdx)
        words.push(word)
        startIdx = spaces[i];
        endIdx = spaces[i + 1];
    }
    
    return words.join(" ")
};